### ISSUE TITLE
More shortcut keys, in particular for recording

### ISSUE THREAD

**Issue by Aeny202** [mumble-voip/mumble#2705]
I was looking for a shortcut key to start/stop recording in Mumble but couldn't find any. I then proceeded to check the latest 1.3 version and saw that it wasn't implemented there either. Did a quick check on here and couldn't find an issue open for this already. 

Use case: 
Me and a couple of friends use mumble for our game play recording, as it is now we have to tab out of the game, hit record, tab back in game, then later sync it up with the video capture software. I think it would be a decent enhancement if this could be done with a hotkey, we could map it to the same key as the video capture software, thus getting rid of several in between steps like syncing afterwards and tabbing out. Some games do not like to be tabbed.

**Comment by Kissaki** [comment]
The current recording functionality (UI) resides in a dialog which also configures how and where to save the recordings. How would you expect those to be handled if you just press a button to record?

It seems our dialog already keeps the settings. So I guess apart from the settings not being immediately visible, it makes sense to use them. It may also make sense to move them to the settings dialog, as it is no longer necessary to open the dialog and hit start.

**Comment by mkrautz** [comment]
Additional data for the shortcut could define the path to save to. Or we could use the value already set in the recording dialog.

**Comment by Aeny202** [comment]
The record dialog always seemed like an afterthought to me. I'm sure whoever added it had their reasons to do it the way it is, but I can't see it. How this would be implemented is up to the pro's.

ui/program design is not something I'm well versed in. As long as none of the settings (path, output format, multi channel) are lost then I'm a happy camper.

**Comment by mkrautz** [comment]
Discussed this on IRC, and we have a design we can move forward with:

	08:58:05 <fyre> anyway to set up a hotkey for recording?
	09:22:19 <@mkrautz> fyre: not currently... I guess ideally there'd be separate start and stop shortcuts, instead of a toggle...?
	09:22:58 <fyre> well either or really
	09:23:33 <fyre> as long as mumble creates a new file each time you hit start
	09:24:21 <fyre> having this will save me setting up something like voicemeteer and using audacity
	09:25:04 <fyre> well its technically possible to do it right now, but you have to open the recording window everytime
	09:25:23 <fyre> tbh i am surprised its not there as an option, ts3 has had it for donkey years
	09:28:47 <@mkrautz> how would an ideal design look to you? from what you wrote earlier, it seems like you want a 'start recording' shortcut that: 1) if not recording, starts recording 2) if already recording, stops the existing recording and starts a new one
	09:29:25 <fyre> 2 not really needed, a toggle will do
	09:29:38 <fyre> never encountered a system like 2

Basically:

   - The shortcut will use the settings you've configured in the recorder dialog.
   - There will be a "Start/stop Recording" shortcut. A toggle.

Unfortunately, we need to refactor VoiceRecorder a bit to be able to move start/stop to a shortcut.

**Comment by mkrautz** [comment]
If anyone feels like implementing this, go ahead. I think the design above is sensible.

**Comment by QuickMick** [comment]
when is this "sometime" happening?

**Comment by thebushyone** [comment]
Hi Guys,

Did anything ever progress with this?

Thanks

**Comment by Krzmbrzl** [comment]
Not really. There was https://github.com/mumble-voip/mumble/pull/6260 that added a shortcut to open the recording dialog. Not quite what is asked for in this issue, but maybe still an improvement?

**Comment by Bullsaur** [comment]
Looking at #6260, it doesn't quite hit what was wanted from the original post.
I'll see about working on this issue as a first-time contributor.

**Comment by serebryakov02** [comment]
Hello,

I looked through this issue and the related PR #6260 before starting on it.

My understanding is that #6260 added a global shortcut to open the recording dialog, but it does not implement the original request here: a shortcut that directly starts/stops recording, using the settings already configured in the recorder dialog.

I also saw the earlier proposed design: add a “Start/stop Recording” toggle shortcut that uses the current recorder settings. That still sounds like the intended direction to me.

Is this still relevant, and is anyone actively working on it? If not, I would like to pick this up.

**Comment by Hartmnt** [comment]
> Is this still relevant, and is anyone actively working on it? If not, I would like to pick this up.

AFAIK this issue is up for grabs

**Comment by serebryakov02** [comment]
@Hartmnt Thanks for confirming. I would like to work on this issue.

Could this be assigned to me, please? I will start looking into the implementation and will open a PR once I have something ready.
